wgpu-hal/src/metal/mod.rs

```diff
@@ -108,13 +108,14 @@ impl crate::Instance<Api> for Instance {
             .into_iter()
             .map(|dev| {
                 let name = dev.name().into();
+                let has_unified_memory = dev.has_unified_memory();
```

Comment:
We can't call it unconditionally, this method is only available on macOS 10.15 and iOS 13.0.
Generally what we do for this is adding a boolean into `PrivateCapabilities` and checking it here.

Reply:
I added an `Option<bool>` so that I could tell whether it was set or not. That way, when the OS versions are new enough, I can set and use `has_unified_memory` to indicate whether the GPU is integrated or not, and for older OSes where the API is unavailable, I can fall back to using the low power property to indicate an integrated GPU.